tests/lib/rules/jsx-no-undef.js

```diff
@@ -29,8 +27,6 @@ const parserOptions = {
 // -----------------------------------------------------------------------------
 
 const ruleTester = new RuleTester({ parserOptions });
-const linter = ruleTester.linter || eslint.linter || eslint.Linter;
-linter.defineRule('no-undef', require('../../helpers/getESLintCoreRule')('no-undef'));
```

Comment:
There's no need to define a core rule, it's already available in the linter.

Reply:
hm, there must be some reason this was here before - is there a chance that in an older version of eslint, it was necessary? (we support down to v2)

Reply:
I believe `invalid` test cases would be failing otherwise. But to avoid concerns, I'll change this to retain the previous code for ESLint < 9.

Reply:
Updated in https://github.com/jsx-eslint/eslint-plugin-react/pull/3759/commits/5edc3954a86fc6fc68501b1cded5a4ed9db8c32a